app/mailers/host_report_mailer.rb

```diff
@@ -0,0 +1,52 @@
+require 'uri'
+
+class HostReportMailer < ApplicationMailer
+  helper ApplicationHelper
+
+  # sends out a summary email of hosts and their metrics (e.g. how many changes failures etc).
+  def summary(options = {})
+    raise ::Foreman::Exception, N_("Must specify a valid user with email enabled") unless (user = User.find(options[:user]))
+    hosts = User.as user do
+      Host::Managed.authorized_as(user, :view_hosts, Host).order(:last_report).limit(100)
+    end
+
+    @format = options[:format]
+    time = options[:time] || 1.day.ago
+    host_data = HostReport.summarise(time, hosts, @format)
+    total_metrics = load_metrics(host_data)
```

Comment:
What this method does? You generate a hash with `summarise` and then transform it into other hash in `load_metrics`. That is not readable, just generate the data structure for the view correctly in the first method.

Also simplify it, you can decide whatever you like in this structure, change the view (the template), no need to use `:metric` at all. I suggest to cleanup the data structure and make it as close as it looks in the email HTML.